stdlib/Mmap/src/Mmap.jl

```diff
@@ -191,7 +191,7 @@ function mmap(io::IO,
     isopen(io) || throw(ArgumentError("$io must be open to mmap"))
     isbitstype(T)  || throw(ArgumentError("unable to mmap $T; must satisfy isbitstype(T) == true"))
 
-    len = prod(dims) * sizeof(T)
+    len = prod(dims .|> big) * sizeof(T)
```

Comment:
I think using `big` here will allocate unnecessarily, can this be done with something like `reduce(checked_mul, dims)` or using `mul_with_overflow` (checking the returned flag manually)?

If overflow protection is the goal, this should also take the final multiplication with `sizeof(T)` into account, so maybe an explicit loop with `sizeof(T)` as the initial value is best?